sockets/utils/getSocketUrlParts.js

```diff
@@ -82,6 +82,15 @@ function getSocketUrlParts(resourceQuery) {
   pathname = parsedQuery.sockPath || pathname;
   port = parsedQuery.sockPort || port;
 
+  if (!hostname || !pathname || !port) {
+    throw new Error(
+      '[React Refresh] Failed to get an URL for the socket connection. ' +
+        'This might occur when the execution script doesn’t have `src` attribute set. ' +
+        'Either specify it or use overlay options ' +
```

Comment:
This seems a bit difficult to action to me. What exactly are we expecting the user to do?

Reply:
good point.

Maybe we could say something like 

`You can specify sockPort, sockPath, and sockHost in either webpack-dev-server's config, or in react-refresh-webpack-plugin's config` or something like that?